src/Propel/Runtime/Validator/Constraints/Date.php

```diff
@@ -15,7 +15,7 @@ class Date extends SymfonyDateConstraint
     /**
      * @var string
      */
-    public $message = 'This value is not a valid date.';
+    public string $message = 'This value is not a valid date.';
```

Comment:
I think this doesn't work in Symfony < 7. That's what i meant by my comment in #1990. But maybe you can also somehow make it depend on the Symfony version?

Reply:
Thanks for the suggestion and the quick review. I see what you mean now. I will try after work or in the weekend to address your comments and to see if I can find a suitable solution for this one specifically.

Reply:
Sure! Glad you jumped into action :-) 
I also tried a few things, you can check them in pableu/Propel2#1. It's kinda hacky, but the pipeline works with all Symfony and PHP versions **and** static analysis. You can take whatever inspiration from my branch :-)

Reply:
@pableu thank you very much for your help and the the work on your branch, it helped a lot. 
For the date $message property, I simply removed it, as the property in the parent was the same.
For the File loader class, for the load method, I narrowed the return type from mixed to a nullable array, and returned null.

Reply:
> For the date $message property, I simply removed it, as the property in the parent was the same.

Perfect 🤣

Thanks for your work, let's hope it can get merged soon!